functions/Read-DbaBackupHeader.ps1

```diff
@@ -101,99 +101,166 @@ function Read-DbaBackupHeader {
     )
 
     begin {
-        $loopCnt = 1
+        Write-Message -Level InternalComment -Message "Starting reading headers"
         try {
             $server = Connect-SqlInstance -SqlInstance $SqlInstance -SqlCredential $SqlCredential
```

Comment:
This `$server` never used later, especially why its not used instead of `$SqlInstance` at line 120

Reply:
This is because of the runspace @Zhunya at line 210 you can see the transfer of the variables so the runspace threads are being passed that `$server` object.